safe-global/safe-wallet-web
# feat: Add email section to settings page and button to retrieve email

## What it solves

Resolves #3291
Resolves #2923 

## How this PR fixes it

- Adds a new email section below the recovery section
- Adds a "Sign to view" button that fetches the CGW endpoint to retrieve email for the current signer

## Out-of-scope

- Adding a form to add or edit an email

## How to test it

1. Open a safe that has recovery enabled
2. Observe the new section on the settings page
3. Press the Sign to view button
4. Sign the message
5. Observe the button is replaced with either the email or a message that there is no email

## Screenshots
<img width="1257" alt="Screenshot 2024-02-23 at 11 55 10" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/acfdd33e-893d-4ed4-8ad0-c34393873303">
<img width="1266" alt="Screenshot 2024-02-23 at 11 55 20" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/14ac56a9-9d3f-4a06-ad61-4739b79f7ec7">

## Checklist
